src/components/organisms/DustStorm/DustStorm.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useCallback, useEffect } from "react";
+import "./DustStorm.scss";
+import { WithId } from "utils/id";
+import { AnyVenue } from "types/Firestore";
+import firebase from "firebase";
+import { useHistory } from "react-router-dom";
+import { OnlineStatsData } from "../../../../src/types/OnlineStatsData";
+
+const getRandomInt = (max: number) => {
+  return Math.floor(Math.random() * Math.floor(max + 1));
+};
+
+interface PoLuckButtonProps {
+  openVenues?: Array<WithId<AnyVenue>>;
+  afterSelect: () => void;
+}
+
+const PotLuckButton: React.FC<PoLuckButtonProps> = ({
+  openVenues,
+  afterSelect,
+}) => {
+  const history = useHistory();
+  const goToRandomVenue = useCallback(() => {
+    const ExperiencesOrArtpieces = openVenues?.filter(
+      (venue) => venue.template === "zoomroom" || venue.template === "artpiece"
+    );
+
+    if (!ExperiencesOrArtpieces) return;
+
+    const randomVenue =
+      ExperiencesOrArtpieces[getRandomInt(ExperiencesOrArtpieces?.length - 1)];
+    afterSelect();
+
+    if (randomVenue?.template === "artpiece")
+      history.push(`/in/${randomVenue.id}`);
+    if (randomVenue?.template === "zoomroom")
+      window.open(`${randomVenue.zoomUrl}`);
+  }, [openVenues, afterSelect, history]);
+  if (!openVenues) {
+    return <></>;
+  }
+  return (
+    <button onClick={goToRandomVenue} className="btn btn-primary">
+      Join the closest venue
+    </button>
+  );
+};
+
+export const DustStorm = () => {
+  const [openVenues, setOpenVenues] = useState<OnlineStatsData["openVenues"]>(
+    []
+  );
+
+  useEffect(() => {
+    const getOnlineStats = firebase
+      .functions()
+      .httpsCallable("stats-getOnlineStats");
+    const updateStats = () => {
+      getOnlineStats()
+        .then((result) => {
+          const { openVenues } = result.data as OnlineStatsData;
+          setOpenVenues(openVenues);
+        })
+        .catch(() => {}); // REVISIT: consider a bug report tool
+    };
+    updateStats();
+    const id = setInterval(() => {
+      updateStats();
+    }, 60 * 1000);
```

Comment:
Should’ve done this sooner, but can we please increase the interval to 5 min here.. we may overload the system otherwise..